### PR TITLE
Remove DES_check

### PR DESCRIPTION
**Pull request by richsalz**
This is old, hoary, code.  And having an optimization to check for weak keys -- in DES no less -- set by a global variable... times have moved past this kind of hack.

### CODE REVIEW

**Comment by paulidale** [comment]
A rebase to master to get rid of the conflict would be appreciated.

**Comment by richsalz** [comment]
rebased and pushed.

**Comment by paulidale** [comment]
Thanks for that, merged to master.

File: include/openssl/des.h [hunk 1]
```diff
@@ -148,8 +145,7 @@ int DES_check_key_parity(const_DES_cblock *key);
 int DES_is_weak_key(const_DES_cblock *key);
 /*
  * DES_set_key (= set_key = DES_key_sched = key_sched) calls
- * DES_set_key_checked if global variable DES_check_key is set,
- * DES_set_key_unchecked otherwise.
+ * DES_set_key_unchecked
```
Comment (t8m) [inline include/openssl/des.h]: This comment is wrong. It now calls DES_set_key_checked.

**Comment by t8m** [comment]
This has a potential of breaking badly legacy code that does not expect the DES_set_key to fail. As the default always was to not check, it should be kept so the DES_set_key() should be just DES_set_key_unchecked().

File: include/openssl/des.h (hunk 1 above)
Comment (richsalz) [reply include/openssl/des.h]: oops.  fix in #9292

**Comment by richsalz** [comment]
if the code is still using single-DES it has more problems than just checking the key. :)
But the API didn't change, so any code that knew it could not fail was also cheating by looking at the source.

**Comment by levitte** [comment]
> if the code is still using single-DES it has more problems than just checking the key. :)
> But the API didn't change, so any code that knew it could not fail was also cheating by looking at the source.

Uhmmmmm, you removed its declaration from a public header file, so technically speaking, the API did change.

That being said, this was a really old thing that should have been irrelevant long ago...

**Comment by richsalz** [comment]
By "its declaration" you must mean the global variable to toggle the behavior.  Well, yeah, that was kinda the point. :)

**Pull request closed without merging**